File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/fixtures/php/callbacks.php
<?php
class Router {
    private $primary;
    private $secondary;
    function setPrimary($cb) { $this->primary = $cb; } // store_primary
    function setSecondary($cb) { $this->secondary = $cb; } // store_secondary
    function firePrimary() { ($this->primary)(); } // call_primary
    function fireSecondary() { ($this->secondary)(); } // call_secondary
}
class Other {
    private $primary;
    function firePrimary() { ($this->primary)(); } // call_other
}

File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/fixtures/lua/callbacks.lua
local Router = {}
function Router:setPrimary(cb) self.primary = cb end -- store_primary
function Router:setSecondary(cb) self.secondary = cb end -- store_secondary
function Router:firePrimary() self.primary() end -- call_primary
function Router:fireSecondary() self.secondary() end -- call_secondary
local Other = {}
function Other:firePrimary() self.primary() end -- call_other
return {Router = Router, Other = Other}
